src/libsystemd/sd-netlink/sd-netlink.c

```diff
@@ -947,15 +947,15 @@ _public_ int sd_netlink_add_match(
                                           destroy_callback, userdata, description);
 }
 
-_public_ int sd_netlink_attach_filter(sd_netlink *nl, size_t len, struct sock_filter *filter) {
+_public_ int sd_netlink_attach_filter(sd_netlink *nl, size_t len, const struct sock_filter *filter) {
         assert_return(nl, -EINVAL);
         assert_return(len == 0 || filter, -EINVAL);
 
         if (setsockopt(nl->fd, SOL_SOCKET,
                        len == 0 ? SO_DETACH_FILTER : SO_ATTACH_FILTER,
                        &(struct sock_fprog) {
                                .len = len,
-                               .filter = filter,
+                               .filter = (struct sock_filter*) filter,
```

Comment:
Maybe better to constify `struct sock_fprog`??

Reply:
I tried that, but gcc still emits the error. I think the problem is that `filter` is a non-`const` pointer, so even if the structure *is* `const`, we get the warning.

Reply:
Hm, spurious. But OK.